hecuba_core/src/py_interface/UnitParser.cpp

```diff
@@ -299,4 +301,262 @@ PyObject *UuidParser::c_to_py(const void *payload) const {
     CassUuid uuid = {*((uint64_t *) it), *((uint64_t *) it + 1)};
     cass_uuid_string(uuid, final);
     return PyString_FromString(final);
+}
+
+TupleParser::TupleParser(const ColumnMeta &CM) : UnitParser(CM) {
+    this->col_meta = CM;
+}
+
+
+int16_t TupleParser::py_to_c(PyObject *obj, void *payload) const {
+    if (obj == Py_None) throw ModuleException("Error parsing PyObject from py to c, expected a non-none object");
+    if (!PyTuple_Check(obj)) throw ModuleException("Error parsing PyObject from py to c, expected a tuple object");
+    if (PyTuple_Size(obj) != col_meta.pointer->size())
+        throw ModuleException(
+                "Error parsing PyObject from py to c, expected size of Py_tuple being the same as Column_meta");
+    uint32_t total_malloc = 0;
+    for (int i = 0; i < col_meta.pointer->size(); ++i) {
+        total_malloc = total_malloc + col_meta.pointer->at(i).size;
+    }
+    void *internal_payload = malloc(total_malloc);
+    uint32_t size = (uint32_t) PyTuple_Size(obj);
+    for (uint32_t i = 0; i < size; ++i) {
+        PyObject *tuple_elem = PyTuple_GetItem(obj, i);
+        if (tuple_elem == Py_None) throw ModuleException(
+                    "Error parsing PyObject from py to c, expected a non-none object at position " + std::to_string(i) +
+                    " in Py_tuple");
+        CassValueType cvt = this->col_meta.pointer->at(i).type;
+        void *destiny = (char *) internal_payload + this->col_meta.pointer->at(i).position;
+        switch (cvt) {
+            case CASS_VALUE_TYPE_VARCHAR:
+            case CASS_VALUE_TYPE_TEXT:
+            case CASS_VALUE_TYPE_ASCII: {
+                TextParser tp = TextParser(col_meta.pointer->at(i));
+                tp.py_to_c(tuple_elem, destiny);
+                break;
+            }
+            case CASS_VALUE_TYPE_VARINT:
+            case CASS_VALUE_TYPE_BIGINT: {
+                Int64Parser i64p = Int64Parser(col_meta.pointer->at(i));
+                i64p.py_to_c(tuple_elem, destiny);
+                break;
+            }
+            case CASS_VALUE_TYPE_BLOB: {
+                BytesParser bp = BytesParser(col_meta.pointer->at(i));
+                bp.py_to_c(tuple_elem, destiny);
+                break;
+            }
+            case CASS_VALUE_TYPE_BOOLEAN: {
+                BoolParser bp = BoolParser(col_meta.pointer->at(i));
+                bp.py_to_c(tuple_elem, destiny);
+                break;
+            }
+                //TODO parsed as uint32 or uint64 on different methods
+            case CASS_VALUE_TYPE_COUNTER: {
+                Int64Parser i64p = Int64Parser(col_meta.pointer->at(i));
+                i64p.py_to_c(tuple_elem, destiny);
+                break;
+            }
+            case CASS_VALUE_TYPE_DECIMAL: {
+                //TODO
```

Comment:
Remove the cases not implemented, and put a default one which raises ModuleException("Support for this data type not implemented")